Applications/CMC/cmc.cpp

```diff
@@ -111,20 +116,36 @@ int main(int argc,char **argv)
 
     // CONSTRUCT
     cout<<"Constructing investigation from setup file "<<setupFileName<<".\n"<<endl;
-    CMCTool cmcgait(setupFileName);
-
-    // PRINT MODEL INFORMATION
-    Model& model = cmcgait.getModel();
-    cout<<"-----------------------------------------------------------------------\n";
-    cout<<"Loaded library\n";
-    cout<<"-----------------------------------------------------------------------\n";
-    model.printBasicInfo(cout);
-    cout<<"-----------------------------------------------------------------------\n\n";
-
-
-    // RUN
-    cmcgait.run();
-
+    if(specifiedMaxNumThreads != -1)
+    {
+        CMCTool cmcgait(setupFileName,specifiedMaxNumThreads);
+        
```

Comment:
> (Wouldn't it go out of scope?)

Yes.


You can either make a member function on the tool for the number of threads, or you can use a pointer:

std::unique_ptr<CMCTool> cmcgait;
if (...) { cmcgait.reset(new CMCTool(...)) } else { ...} cmcgait->run();